frontend/javascripts/oxalis/model/sagas/proofread_saga.ts

```diff
@@ -0,0 +1,451 @@
+    nodePositionsInSurround.map((nodePosition) =>
+      call(
+        [api.data, api.data.getRawDataCuboid],
+        fallbackLayerName,
+        nodePosition,
+        V3.add(nodePosition, mag),
+      ),
+    ),
+  );
+  // TODO HACK: This only works for uint32 segmentations
+  const segmentIdsInSurround = segmentIdsArrayBuffers.map((buffer) => new Uint32Array(buffer)[0]);
+
+  if (oldSegmentIdsInSurround != null) {
+    const segmentIdsInSurroundSet = new Set(segmentIdsInSurround);
+    // Remove old meshes in oversegmentation
+    yield* all(
+      oldSegmentIdsInSurround.map((nodeSegmentId) =>
+        // Only remove meshes that are not part of the new segment surround
+        segmentIdsInSurroundSet.has(nodeSegmentId)
+          ? null
+          : put(removeIsosurfaceAction(layerName, nodeSegmentId)),
+      ),
+    );
+  }
+
+  oldSegmentIdsInSurround = [...segmentIdsInSurround];
+
+  /* Load fine ad hoc meshes of the segments in the oversegmentation in the click position surround */
+
+  const noMappingInfo = {
+    mappingName: null,
+    mappingType: null,
+    useDataStore: true,
+  };
+  const oldPreferredQuality = yield* select(
+    (state) => state.temporaryConfiguration.preferredQualityForMeshAdHocComputation,
+  );
+  yield* put(
+    updateTemporarySettingAction(
+      "preferredQualityForMeshAdHocComputation",
+      proofreadFineResolutionIndex(),
+    ),
+  );
+  yield* all(
+    segmentIdsInSurround.map((nodeSegmentId, index) =>
+      put(
+        loadAdHocMeshAction(
+          nodeSegmentId,
+          nodePositionsInSurround[index],
+          noMappingInfo,
+          layerName,
+        ),
+      ),
+    ),
+  );
+  yield* put(
+    updateTemporarySettingAction("preferredQualityForMeshAdHocComputation", oldPreferredQuality),
+  );
+}
+
+function* createEditableMapping(): Saga<void> {
+  const tracingStoreUrl = yield* select((state) => state.tracing.tracingStore.url);
+  // Save before making the mapping editable to make sure the correct mapping is activated in the backend
+  yield* call([Model, Model.ensureSavedState]);
+  // Get volume tracing again to make sure the version is up to date
+  const upToDateVolumeTracing = yield* select((state) => getActiveSegmentationTracing(state));
+  if (upToDateVolumeTracing == null) return;
+
+  const volumeTracingId = upToDateVolumeTracing.tracingId;
+  const layerName = volumeTracingId;
+  const serverEditableMapping = yield* call(makeMappingEditable, tracingStoreUrl, volumeTracingId);
+  // The server increments the volume tracing's version by 1 when switching the mapping to an editable one
+  yield* put(setVersionNumberAction(upToDateVolumeTracing.version + 1, "volume", volumeTracingId));
+  yield* put(setMappingNameAction(layerName, serverEditableMapping.mappingName));
+  yield* put(setMappingisEditableAction());
+  yield* put(initializeEditableMappingAction(serverEditableMapping));
+}
+
+function* splitOrMergeAgglomerate(action: MergeTreesAction | DeleteEdgeAction) {
```

Comment:
this saga is quite long. would it be possible to split it up? or is this too tedious since a lot of variables would need to be passed?